gc/base/ParallelDispatcher.cpp

```diff
@@ -457,6 +457,26 @@ MM_ParallelDispatcher::recomputeActiveThreadCountForTask(MM_EnvironmentBase *env
 	 * available and ready to run).
 	 */
 	uintptr_t taskActiveThreadCount = OMR_MIN(_activeThreadCount, threadCount);
+
+	/* Account for Adaptive Threading. Ignore if
+	 *
+	 *  1) User forced a thread count (e.g Xgcthreads)
+	 *  2) Adaptive threading flag is not set (-XX:-AdaptiveGCThreading)
+	 *  3) The task wasn't recommended a thread count (currently only recommended for STW Scavenge Tasks)
+	 */
+	if (_extensions->adaptiveThreadigEnabled() && task->getRecommendedWorkingThreads() != UDATA_MAX) {
+		/* Bound the recommended thread count. Determine the  upper bound for the thread count,
+		 * either the user specified adaptiveThreadCount (-XXgc:adaptiveThreads=) or else default max
+		 */
+		uintptr_t upperThreadBound = (_extensions->adaptiveThreadCount == 0) ? _threadCountMaximum : _extensions->adaptiveThreadCount;
+		taskActiveThreadCount = OMR_MIN(upperThreadBound, task->getRecommendedWorkingThreads());
+		taskActiveThreadCount = OMR_MAX(1, taskActiveThreadCount);
```

Comment:
perhaps it's cleaner if we make getRecommendedWorkingThreads returns >= 1 (if that's the reason we do this  OMR_MAX(1,...)

Reply:
recommended thread will always return  >= 1,
https://github.com/RSalman/omr/blob/893eb8f9413eb74eecdd8135ea4b62e115ab019f/gc/base/standard/Scavenger.cpp#L567-L569

 so this isn't needed